lib/mtl/rails/card_file_presenter.rb

```diff
@@ -56,8 +56,14 @@ def delete(params)
         view.mtl_icon :close, class: 'close', data: data.reject { |_, v| v.blank? }
       end
 
-      def modal(filename, params)
-        { mtl_document_modal: 'open', mtl_document_name: filename } if params.delete(:modal)
+      def data(filename, params)
+        data = params[:data] || {}
+        data = data.merge(modal_data(filename)) if params[:modal]
+        data
+      end
+
+      def modal_data(filename)
```

Comment:
If you split it into two methods, then push the decision as well, e.g. either

```ruby
data = data.merge(mtl_document_modal: 'open', mtl_document_name: filename) if params[:modal]
```

or

```ruby
def modal_data(filename, params)
  return {} unless params[:modal]
  { ... }
end
```